### ISSUE TITLE
should not check no permission folder

### ISSUE THREAD

**Issue by lovemyliwu** [facebook/pyre-check#81]
config file:
```
{
  "binary": "/home/smite/.local/bin/pyre.bin",
  "search_path": [".", "/usr/local/lib/python3.6/dist-packages/"],
  "source_directories": [
    "common",
    "applications"
  ],
  "do_not_check": [
    "*_dev",
    "applications/mock_api/*",
    "applications/web/*",
    "*_models.py",
    "*_model.py",
    "*_models_*.py",
    "*/__init__.py",
    "*/apps.py",
    "*/admin*.py",
    "*/urls.py",
    "tests/conftest.py",
    "*/management/*"
  ],
  "typeshed": "/home/smite/.local/lib/pyre_check/typeshed/"
}
```

but when I run `pyre check`, it raise a non-relative folder permission check failure!

```
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/auth`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/users`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/licenses`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/splunk_management_console/lookups`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/learned/local`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/search/local`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/splunk_archiver/java-bin`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/run`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/log`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/spool`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/lib`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/application_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/interface_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/rule_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/default`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/crawl_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/mysql`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/cleandata_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/performance_schema`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/auth`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/users`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/licenses`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/splunk_management_console/lookups`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/learned/local`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/search/local`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/splunk_archiver/java-bin`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/run`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/log`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/spool`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/lib`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/application_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/interface_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/rule_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/default`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/crawl_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/mysql`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/cleandata_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/performance_schema`
 ƛ Could not parse 23 external files due to syntax errors! Run with `--show-parse-errors` for more details.
 ƛ No type errors found

```

any idea?

**Comment by lovemyliwu** [comment]
@dark see below
```
smite:~/work/quantgroup_loan_v2>pwd
/home/smite/work/quantgroup_loan_v2
smite:~/work/quantgroup_loan_v2>pyre check
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/auth`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/users`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/licenses`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/splunk_management_console/lookups`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/learned/local`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/search/local`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/splunk_archiver/java-bin`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/run`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/log`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/spool`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/lib`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/application_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/interface_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/rule_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/default`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/crawl_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/mysql`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/cleandata_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/performance_schema`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/auth`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/users`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/licenses`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/splunk_management_console/lookups`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/learned/local`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/search/local`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/data/apps/splunk_archiver/java-bin`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/run`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/log`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/spool`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/splunk_dev/conf/lib`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/application_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/interface_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/rule_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/default`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/crawl_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/mysql`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/cleandata_db`
 ƛ Could not list `/home/smite/work/quantgroup_loan_v2/mysql_dev/data/performance_schema`
 ƛ Could not parse 23 external files due to syntax errors! Run with `--show-parse-errors` for more details.
 ƛ No type errors found
smite:~/work/quantgroup_loan_v2>echo $?
1
smite:~/work/quantgroup_loan_v2>pyre rage
 ƛ Shared source directory is stale, updating...
Displaying logs for Watchman watched directories:


```

**Comment by lovemyliwu** [comment]
ok, so can we add flag e.g. **ignore_path** to manage the absolute not related directory?

**Comment by dark** [comment]
We haven't added a flag like that yet, because we have not found a compelling use case for it.
Since it would complicate the logic quite a bit, we are not planning to add one either for now.

**Comment by dark** [comment]
For the record, we have reconsidered our decision, and this feature has been implemented in `v0.0.18` (due soon).
